packages/runner/lib/exec.ts

```diff
@@ -54,7 +54,19 @@ export async function exec(nangoProps: NangoProps, isInvokedImmediately: boolean
                 return await scriptExports.default(nango);
             }
         }
-    } catch (error) {
-        throw new Error(`Error executing code '${error}'`);
+    } catch (error: any) {
+        if (error instanceof ActionError) {
+            const { type, payload } = error;
+            return {
+                success: false,
+                error: {
+                    type,
+                    payload
+                },
+                response: null
+            };
+        } else {
+            throw new Error(`Error executing code '${error}'`);
+        }
```

Comment:
why not simply throwing `error` in the runner and deal with the response format in `integration.service`. This way it is all in the same place (aka: in the try catch in integration.service)

Reply:
This was existing already so I just kept this as it was: https://github.com/NangoHQ/nango/blob/master/packages/runner/lib/exec.ts#L58